src/client/components/MealId.js

```diff
@@ -0,0 +1,290 @@
+            </div>
+            <h5 className="mb-3 add-review-title">Add your review</h5>
+            <AddReview mealId={id} />
+        </div>
+    )
+}
+
+function AddReview(props) {
+    const [newReview, setNewReview] = useState({})
+    const [inputReview, setInputReview] = useState({
+        title: "",
+        meal_id: props.mealId,
+        description: "",
+        stars: ""
+    })
+    function postNewReservation() {
+        (async () => {
+            await fetch('http://localhost:5000/api/reviews', {
+                method: 'POST', // *GET, POST, PUT, DELETE, etc.
+                mode: 'cors', // no-cors, *cors, same-origin
+                headers: {
+                    'Content-Type': 'application/json',
+                    // 'Content-Type': 'application/x-www-form-urlencoded',
+                },
+                body: JSON.stringify(newReview) // body data type must match "Content-Type" header
+            })
+        })();
+
+    }
+
+    useEffect(() => {
+        setNewReview(inputReview)
+    }, [inputReview])
+
+    function handleChange(input) {
+        const { name, value } = input;
+        setInputReview((prev) => {
+            return {
+                ...prev,
+                [name]: value
+            }
+        })
+    }
+
+    function handleSubmit() {
+        postNewReservation();
+        setInputReview({
+            title: "",
+            meal_id: props.mealId,
+            description: "",
+            stars: ""
+        })
+    }
+    return (
+        <div className="reservation-form">
+            <Form onSubmit={handleSubmit}>
+                <TextInput change={handleChange} label="Title" placeholder="Enter a title" name="title" value={inputReview.title} />
+                <TextArea change={handleChange} label="Description" placeholder="Describe your experience" name="description" value={inputReview.description} />
+                <NumberInput change={handleChange} label="Stars" placeholder="Review 1-5" name="stars" value={inputReview.stars} min="1" max="5" />
+                <Button type="submit" className="add-button">
+                    Add review
+                </Button>
+            </Form>
+        </div>
+    )
+}
+
+function MealForms(props) {
+    return (
+        <Accordion className="accordion-box">
+            <Card>
+                <Card.Header >
+                    <Accordion.Toggle as={Button} variant="link" eventKey="0" className="text-uppercase">
+                        Make a reservation
+                    </Accordion.Toggle>
+                </Card.Header>
+                <Accordion.Collapse eventKey="0">
+                    <Card.Body>
+                        {props.visible ? <AddNewReservation mealId={props.id} guests={props.availableMeal} /> : <div>No available reservations</div>}
+                    </Card.Body>
+                </Accordion.Collapse>
+            </Card>
+        </Accordion>
+    )
+}
+
+function MealData(props) {
```

Comment:
Also here. The word "Data" is the same as "handle" it is a "noise word" that could refer to anything